components/raftstore/src/store/fsm/peer.rs

```diff
@@ -931,16 +931,37 @@ where
     // the flashback-only command in this way, But for RW local reads which need
     // to be considered, we let the leader lease to None to ensure that local reads
     // are not executed.
-    fn on_prepare_flashback(&mut self, ch: Sender<bool>) {
+    fn on_prepare_flashback(&mut self, version: u64, ch: Sender<Result<()>>) {
+        let region_id = self.region().get_id();
         info!(
             "prepare flashback";
-            "region_id" => self.region().get_id(),
+            "region_id" => region_id,
             "peer_id" => self.fsm.peer.peer_id(),
         );
+        // Check if there is another flashback in in progress.
         if self.fsm.peer.flashback_state.is_some() {
-            ch.send(false).unwrap();
+            ch.send(Err(Error::Other(
+                format!("region {} is in another flashback progress", region_id).into(),
+            )))
+            .unwrap();
             return;
         }
+        // Check if the flashback version is greater than the region safe ts, i.e, the
+        // resolved ts.
+        let meta = self.ctx.store_meta.lock().unwrap();
+        if let Some(safe_ts) = meta.region_read_progress.get_safe_ts(&region_id) {
```

Comment:
what's the relation of safe_ts with resolved_ts

Reply:
It's the same thing actually.

Reply:
but the resolved-ts doesn't mean `min-resolved-ts`

Reply:
The `resolved_ts` here is the latest one for the region, it will be advanced periodically or when any write happens, the concept of "minimum" only makes sense when we're talking about store-level `resolved_ts`.

BTW, I opened a PR to clarify the use of terms `safe_ts` and `resolved_ts`, maybe it will help us distinguish between the use of these two slightly different semantics: https://github.com/tikv/tikv/pull/13441.